src/develop/develop.c

```diff
@@ -1016,6 +1016,7 @@ void dt_dev_add_history_item(dt_develop_t *dev, dt_iop_module_t *module, gboolea
   {
     /* signal that history has changed */
     dt_control_signal_raise(darktable.signals, DT_SIGNAL_DEVELOP_HISTORY_CHANGE);
+    dt_control_signal_raise(darktable.signals, DT_SIGNAL_TAG_CHANGED);
```

Comment:
I find this awkward adding DT_SIGNAL_TAG_CHANGED in dt_dev_add_history_item() which is a develop call and as nothing to do with tagging.

Reply:
the problem is that dt_dev_add_history_item already adds tag: `darktable|changed` and since changing `dt_tag_attach_from_gui` (that sends signal on it's own) to `dt_tag_attach` (which doesn't) it is needed to have this signal. This handling is better since it also checks if we have gui... it could be further aguamented to check if `dt_tag_attach` attached tag (otherwise no tag change happened) and not raise the signal ;)

Reply:
yes, I haven't check if the signal raising what necessary or not, I've just added them outside the loops when needed. All other case do the exact same things.
Now I agree that there's room for optimization here : not raising the signal if no tag is really added, for example

Reply:
It's really a bit strange to find the tag setting and raising here. But otherwise it's the lowest code level section we can be sure something has changed and so we are on the safe side.